vlm/src/main/scala/geotrellis/contrib/vlm/RasterSourceRDD.scala

```diff
@@ -115,6 +115,61 @@ object RasterSourceRDD {
     ContextRDD(result, layerMetadata)
   }
 
+  def readFromRDD(
+    readingSourcesRDD: RDD[ReadingSource],
+    layout: LayoutDefinition,
+    partitioner: Option[Partitioner] = None
+  )(implicit sc: SparkContext): MultibandTileLayerRDD[SpatialKey] = {
+    val rasterSourcesRDD = readingSourcesRDD.map { _.source }
+    val summary = RasterSummary.fromRDD(rasterSourcesRDD)
+
+    val cellType = summary.cellType
+
+    val layerMetadata: TileLayerMetadata[SpatialKey] = summary.toTileLayerMetadata(layout, 0)._1
+
+    def getNoDataTile = ArrayTile.alloc(cellType, layout.tileCols, layout.tileRows).fill(NODATA).interpretAs(cellType)
+
+    val maxIndex = readingSourcesRDD.map { _.sourceToTargetBand.values.max }.reduce { _ max _ }
+    val targetIndexes: Seq[Int] = 0 to maxIndex
+
+    val keyedRDD: RDD[(SpatialKey, (Int, Option[MultibandTile]))] =
+      readingSourcesRDD.mapPartitions ({ partition =>
+        partition.flatMap { source =>
+          val layoutSource = LayoutTileSource(source.source, layout)
+
+          layoutSource.keys.flatMap { key =>
+            source.sourceToTargetBand.map { case (sourceBand, targetBand) =>
+              (key, (targetBand, layoutSource.read(key, Seq(sourceBand))))
+            }
+          }
+        }
+      })
+
+    val groupedRDD: RDD[(SpatialKey, Iterable[(Int, Option[MultibandTile])])] =
+      keyedRDD.groupByKey(partitioner.getOrElse(SpatialPartitioner(summary.estimatePartitionsNumber)))
+
+    val result: RDD[(SpatialKey, MultibandTile)] =
+      groupedRDD.mapPartitions ({ partition =>
+        val noDataTile = getNoDataTile
+
+      partition.map { case (key, iter) =>
+        val mappedBands: Map[Int, Option[MultibandTile]] = iter.toSeq.sortBy { _._1 }.toMap
+
+        val tiles: Seq[Tile] =
+          targetIndexes.map { index =>
+            mappedBands.getOrElse(index, None) match {
+              case Some(multibandTile) => multibandTile.band(0)
+              case None => noDataTile
+            }
+          }
+
+        key -> MultibandTile(tiles)
+      }
+    }, preservesPartitioning = true)
+
+    ContextRDD(result, layerMetadata)
+  }
+
   def tiledLayerRDD(
```

Comment:
Should not it be a `tileLayerRDD`?

Reply:
I was thinking `read` since it's dealing with `ReadingSource`s. Regardless, though, the reason I didn't do that, though, is because of:

```
[error] /home/jake/Documents/projects/geotrellis-contrib/vlm/src/main/scala/geotrellis/contrib/vlm/RasterSourceRDD.scala:33:8: in object RasterSourceRDD, multiple overloaded alternatives of method read define default arguments.
[error] object RasterSourceRDD {
```

So I'm going to have to figure out a way to refactor those methods so that they can have the same name.